### PR TITLE
[TOPI] Add dp4a intrinsic to CUDA

### PR DESCRIPTION
**Pull request by vinx13**
Added dp4a intrinsic to TOPI, and refactored gemm_int8 recipe. And then I will send int8 conv2d using dp4a in the next PR.

cc @tqchen @merrymercy

### CODE REVIEW

File: topi/python/topi/cuda/int8_intrinsics.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+"""Int8 intrinsics on CUDA."""
+#pylint: disable=invalid-name
+import tvm
+
+
+def _intrin_dp4a_reduce(x_scope, y_scope, z_scope):
```
Comment (tqchen) [inline topi/python/topi/cuda/int8_intrinsics.py]: let us put it under cuda/tensor_intrin.py and rename to dp4a

File: topi/python/topi/cuda/tensor_intrin.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import tvm
+
+
+def dp4a(x_scope, y_scope, z_scope):
+    """
+    Int8 dot product reduced by every 4 elements using __dp4a
+
+    Parameters
+    ----------
+    x_scope: The storage scope of buffer for lhs
```
Comment (tqchen) [inline topi/python/topi/cuda/tensor_intrin.py]: add data type, see https://docs.tvm.ai/contribute/document.html#document-python

File: topi/python/topi/cuda/tensor_intrin.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+"""Tensor intrinsics on CUDA."""
+#pylint: disable=invalid-name
+import tvm
```
Comment (tqchen) [inline topi/python/topi/cuda/tensor_intrin.py]: try use relative import, to get the function you need

File: topi/python/topi/cuda/tensor_intrin.py (hunk 2 above)
Comment (tqchen) [reply topi/python/topi/cuda/tensor_intrin.py]: Maybe it makes sense to default everything to local

File: topi/python/topi/cuda/tensor_intrin.py (hunk 3 above)
Comment (vinx13) [reply topi/python/topi/cuda/tensor_intrin.py]: @tqchen actually we can't, we are in topi, a different package from tvm

**Comment by tqchen** [comment]
Thanks @vinx13 this is now merged!

**Pull request merged**